### PR TITLE
OCPCLOUD-1675: Add support for generic platforms without failure domains

### PR DESCRIPTION
**Pull request by RadekManak**
This PR allows CPMS to manage control plane machines on all platforms that do not require modification of machine spec for failure domains. 
It does so by creating a generic provider config that contains the spec as RawExtension to allow us to compare machine spec to template spec. 
I've also added resourcebuilder for VSphere for use in test as an example of platform without failure domains.

### CODE REVIEW

File: pkg/machineproviders/providers/openshift/machine/v1beta1/providerconfig/generic_platform.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package providerconfig
+
+import (
+	v1 "github.com/openshift/api/config/v1"
```
Comment (JoelSpeed) [inline pkg/machineproviders/providers/openshift/machine/v1beta1/providerconfig/generic_platform.go:20]: Please import this as `configv1` as per conventions

File: pkg/machineproviders/providers/openshift/machine/v1beta1/providerconfig/providerconfig.go [hunk 2]
```diff
@@ -225,17 +242,37 @@ func (p providerConfig) GCP() GCPProviderConfig {
...
+		"LibvirtMachineProviderConfig":      configv1.LibvirtPlatformType,
+		"OpenStackMachineProviderSpec":      configv1.OpenStackPlatformType,
+		"VSphereMachineProviderSpec":        configv1.VSpherePlatformType,
+		"OvirtMachineProviderSpec":          configv1.OvirtPlatformType,
+		"IBMCloudMachineProviderSpec":       configv1.IBMCloudPlatformType,
+		"KubevirtMachineProviderSpec":       configv1.KubevirtPlatformType,
+		"EquinixMetalMachineProviderConfig": configv1.EquinixMetalPlatformType,
+		"PowerVSMachineProviderConfig":      configv1.PowerVSPlatformType,
+		"AlibabaCloudMachineProviderConfig": configv1.AlibabaCloudPlatformType,
+		"NutanixMachineProviderConfig":      configv1.NutanixPlatformType,
```
Comment (JoelSpeed) [inline pkg/machineproviders/providers/openshift/machine/v1beta1/providerconfig/providerconfig.go:266]: This list is likely to go out of date quite quickly, is there any way we can avoid this? 

What are we using this lookup for exactly?

File: pkg/machineproviders/providers/openshift/machine/v1beta1/providerconfig/providerconfig.go [hunk 3]
```diff
@@ -225,17 +242,37 @@ func (p providerConfig) GCP() GCPProviderConfig {
...
 	}
 
 	platformType, ok := providerSpecKindToPlatformType[kind]
 
+	// Attempt to operate on unknown platforms. This should work if the platform does not require failure domains support.
+	if !ok && (strings.Contains(kind, "MachineProviderConfig") || strings.Contains(kind, "MachineProviderSpec")) {
```
Comment (JoelSpeed) [inline pkg/machineproviders/providers/openshift/machine/v1beta1/providerconfig/providerconfig.go:272]: Why the check for `MachineProviderConfig` or `MachineProviderSpec`?

**Review by JoelSpeed — COMMENTED** [review]
Looks good, few questions and nits reading through

File: pkg/machineproviders/providers/openshift/machine/v1beta1/providerconfig/providerconfig.go (hunk 2 above)
Comment (RadekManak) [reply pkg/machineproviders/providers/openshift/machine/v1beta1/providerconfig/providerconfig.go]: I needed this before I had the default case for unknown platforms. We only need this for platforms that have failure domains.

File: pkg/machineproviders/providers/openshift/machine/v1beta1/providerconfig/providerconfig.go (hunk 3 above)
Comment (RadekManak) [reply pkg/machineproviders/providers/openshift/machine/v1beta1/providerconfig/providerconfig.go]: I wanted a sanity check that it is some kind provider spec, but we can assume it is here and handle the error later if it is invalid.

Comment (JoelSpeed) [reply pkg/machineproviders/providers/openshift/machine/v1beta1/providerconfig/providerconfig.go]: Yeah I think that makes sense, lets not bother here and we can handle other errors later

**Review by JoelSpeed — COMMENTED** [review]
/lgtm
/approve

**Comment by RadekManak** [comment]
Fixed lint

**Pull request merged**